### PR TITLE
optimized some querysets, ie. in the scheduler view. 

### PR DESCRIPTION
**Pull request by christophmeissner**
should now query faster.

### CODE REVIEW

**Comment by christophmeissner** [comment]
Don't know, if the joins and prefetches are an issue for the server, but number of queries got down from ie. 592 to 11 for the scheduler view.

**Comment by chhantyal** [comment]
Wow 592 to 11 is huge reduction. Guess queryset was in loop without prefetch. 

This PR needs update though.

File: shiftmailer/templates/shifts_today.html [hunk 1]
```diff
@@ -4,15 +4,16 @@
...
+    {% if need.registrationprofile_set %}
      {% ifchanged need.topic.title %}
          {{ need.topic.title }}
-         {% endifchanged %}
-     von {{ need.time_period_from.date_time|date:"H" }} - {{ need.time_period_to.date_time|date:"H" }} Uhr angemeldet sind {{ need.get_volunteer_total }}) Freiwillige:
+     {% endifchanged %}
+     von {{ need.time_period_from.date_time|date:"H" }} - {{ need.time_period_to.date_time|date:"H" }} Uhr angemeldet sind {{ need.volunteer_count }}) Freiwillige:
 
-    {% for volunteer in need.get_volunteers %}
+    {% for volunteer in need.registrationprofile_set %}
```
Comment (JannKleen) [inline shiftmailer/templates/shifts_today.html]: didn't test it, but i think this should be ```need.registrationprofile_set.all``` ?

Comment (christophmeissner) [inline shiftmailer/templates/shifts_today.html]: good catch, thanks

**Comment by christophmeissner** [comment]
@pitpalme @maikhoepfel @Doca @chhantyal ping.

**Comment by chhantyal** [comment]
LGTM

**Comment by pitpalme** [comment]
Thanks @chhantyal :+1:

**Comment by christophmeissner** [comment]
yay!

**Comment by maikhoepfel** [comment]
@christophmeissner Great work, thanks! Would've gone about it in a very similar way. Only thing I am not sure about is overriding the default `get_queryset`; I think I might prefer a dedicated method so can still use the less costly query. But given how often we need that extra data, there's no real harm in it either. Let's leave it as it is for now.

**Comment by christophmeissner** [comment]
It's in the admin only. Didn't put it in a manager for exact that reason. ;-)

**Comment by maikhoepfel** [comment]
@christophmeissner Woops, failed the basic reading comprehension test. Excellent.

**Pull request merged**